util/stream.go

```diff
@@ -135,61 +121,107 @@ func (m *MessageStream) outbound() {
 	}
 }
 
+type smallMessageError struct {
+	length int
+}
+
+func (e smallMessageError) Error() string {
+	return fmt.Sprintf("invalid message with length %d is received", e.length)
+}
+
+func newSmallMessageError(length int) *smallMessageError {
+	return &smallMessageError{
+		length: length,
+	}
+}
+
+func getMessageLength(reader *bufio.Reader) (int, error) {
+	msgLength, readErr := func(reader *bufio.Reader) (int, error) {
+		headerBytes, err := reader.Peek(4)
+		if err != nil {
+			return 0, err
+		}
+		messageType := int(headerBytes[1])
+		messageLength := int(binary.BigEndian.Uint16(headerBytes[2:4]))
+		if messageType != 4 {
+			return messageLength, nil
+		}
+		venderHeaderMessageBytes, err := reader.Peek(16)
+		if err != nil {
+			return 0, err
+		}
+		// Check if the message is Type_PacketIn2 or not.
+		experimenterType := binary.BigEndian.Uint32(venderHeaderMessageBytes[12:])
+		if experimenterType != 30 {
```

Comment:
is there no existing constant in the library for this?

Reply:
Yes, we have. But they are defined in package `antrea.io/libOpenflow/openflow15` which has imported the current package. So it may introduce a loop if I directly use the constant.

Reply:
> But maybe we should just redefine local constants in this file?

I would prefer to use another PR to perform such refactor. The change does not only impact this repo but also the consumers.